docs/guides/0X_v8_migration_guide.md

```diff
@@ -0,0 +1,293 @@
+    Note over middle: Custom middlewares, e.g. events publication
+
+    middle ->> start: 
+
+    Note over start: Set response headers
+    Note over start: Serialize result
+
+    start ->> aspnet: 
+
+```
+
+### Fluent validation changes
+
+The `ContextualValidator.RuleForAsync()` was removed due to problems with upgrading to newer `FluentValidation` versions and general impedance with _fluent_ approach. Instead clients should rewrite the validator to use imperative approach with `AbstractValidator.CustomAsync()`.
+
+There is a `IRuleBuilderOptions.AddValidationError()` extension which allows setting CQRS error code to the validation error.
+
+Example old approach:
+
+```csharp
+public class CreateDishCV : ContextualValidator<CreateDish>
+{
+    public CreateDishCV()
+    {
+        this.RuleForAsync(
+            c => c.DishId,
+            (ctx, id) => ctx.GetService<IRepository<Dish>>().FindAsync(id))
+            .Null().WithCode(CreateDish.ErrorCodes.DishAlreadyExists);
+    }
+}
+```
+
+Example new approach:
+
+```csharp
+public class CreateDishCV : ContextualValidator<CreateDish>
+{
+    public CreateDishCV()
+    {
+        RuleFor(c => c.DishId)
+            .CustomAsync((Guid dishId, ValidationContext<CreateDish> ctx, CancellationToken ct) =>
+            {
+                var dish = await ctx.GetService<IRepository<Dish>>().FindAsync(id, ct);
+                if(dish is not null)
+                {
+                  ctx.AddValidationError("Dish already exists", CreateDish.ErrorCodes.DishAlreadyExists);
+                }
+            });
+    }
+}
+```
+
+Removal of `ContextualValidator` class allowed to merge `LeanCode.Validation.Fluent` and `LeanCode.Validation.Fluent.Scoped` projects - lifetime of validators is now a parameter of `IServiceCollection.AddFluentValidation()` method (scoped by default)
+
+## MassTransit changes
+
+`LeanCode.DomainModels.MassTransitRelay` was renamed to `LeanCode.CQRS.MassTransitRelay`
+
+### Registration & startup changes
+
+`MassTransitRelayModule` was removed. Instead use `serviceCollection.AddCQRSMassTransitIntegration()` method. Similarly `MassTransitTestRelayModule` was replaced with `serviceCollection.AddBusActivityMonitor()`
+
+#### Consumer definitions
+
+It's now encouraged to create a separate receive endpoint per each consumer. To configure each consumer use a [consumer definition](https://masstransit.io/documentation/concepts/consumers#definitions). Since most of the consumers will have the same configuration a helper default consumer utility was created. Any consumer that does not have a definition will pick up the default one. Additionally, conventional receive endpoint (queue) name will be just the consumer name - if you want to use full namespace name, you'll need to set the name formatter.
+
+```csharp
+public void ConfigureServices(IServiceCollection services)
+{
+    services.AddCQRSMassTransitIntegration(busCfg =>
+    {
+        busCfg.AddConsumersWithDefaultConfiguration(
```

Comment:
Not a doc issue, but now I think it could be a generic method rather having a type argument, would look nicer 🤷🏻